models/RdfImporter.php

```diff
@@ -22,12 +22,72 @@
 
 namespace oat\taoTestTaker\models;
 
+use common_report_Report as Report;
+use common_Utils;
+use core_kernel_classes_Class;
 use core_kernel_classes_Resource;
+use EasyRdf_Format;
+use EasyRdf_Graph;
+use oat\generis\model\OntologyRdf;
+use oat\generis\model\user\UserRdf;
+use oat\oatbox\log\LoggerService;
+use oat\tao\model\metadata\exception\InconsistencyConfigException;
+use oat\tao\model\TaoOntology;
 use oat\taoTestTaker\models\events\dispatcher\TestTakerImportEventDispatcher;
 use tao_models_classes_import_RdfImporter;
 
 class RdfImporter extends tao_models_classes_import_RdfImporter
 {
+    public const CONFIG_ID = 'taoTestTaker/rdfImporterConfig';
+
+    public const OPTION_STRATEGY = 'strategy';
+
+    public const OPTION_STRATEGY_FAIL_ON_DUPLICATE = 'fail';
+    public const OPTION_STRATEGY_SKIP_ON_DUPLICATE = 'skip';
+    public const OPTION_STRATEGY_IMPORT_ON_DUPLICATE = 'import';
+
+    public const AVAILABLE_STRATEGIES = [
+        self::OPTION_STRATEGY_FAIL_ON_DUPLICATE,
+        self::OPTION_STRATEGY_SKIP_ON_DUPLICATE,
+        self::OPTION_STRATEGY_IMPORT_ON_DUPLICATE
+    ];
+
+    private const NOT_IMPORTABLE_STRATEGIES = [
+        self::OPTION_STRATEGY_FAIL_ON_DUPLICATE,
+        self::OPTION_STRATEGY_SKIP_ON_DUPLICATE
+    ];
+
+    /** @var $strategy string */
+    private $strategy;
+
+    private function readStrategyFromConfig(): string
+    {
+        $config = $this->getServiceLocator()->get(self::CONFIG_ID);
+
+        $strategy = $config->getOption(self::OPTION_STRATEGY);
+
+        if (!in_array($strategy, self::AVAILABLE_STRATEGIES, true)) {
+            $message = sprintf(
+                "Bad strategy `%s` configured. Strategy should be one of `[%s]",
+                $strategy,
+                implode(',', self::AVAILABLE_STRATEGIES)
+            );
+
+            $this->getLogger()->logError($message);
+
+            throw new InconsistencyConfigException($message);
+        }
+
+        $this->strategy = $strategy;
+
+        return $this->getStrategy();
+    }
+
+    private function getStrategy(): string
```

Comment:
It seems this is used only on `readStrategyFromConfig` method. I would remove it